clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3545,6 +3545,30 @@ scope) variables and static local variable as well.
   }];
 }
 
+def OpenCLAddressSpaceGlobalDeviceDocs : Documentation {
+  let Category = DocOpenCLAddressSpaces;
+  let Heading = "[[clang::opencl_global_device]]";
+  let Content = [{
+The global_device address space attribute specifies that an object is allocated
+in global memory on device. It is supposed to be used only in SYCL headers and
```

Comment:
Hm... AFAIK, USM is an OpenCL extensions, which is supported by multiple vendors, so I expect that OpenCL developers might be interested in using these attributes as well.
We can discuss this at today's upsteam meeting.

Reply:
OpenCL spec extension: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/USM/cl_intel_unified_shared_memory.asciidoc

Reply:
> Hm... AFAIK, USM is an OpenCL extensions, which is supported by multiple vendors, so I expect that OpenCL developers might be interested in using these attributes as well.

I agree, this is interesting for other programing model. Wouldn't that also be usable with SVM ? AFAK, it also uses the `global` address space in kernel to address this.

Reply:
> It is intended for use in SYCL headers, and therefore should not appear in OpenCL/SYCL user code.

I wouldn't say that.  As @bader noted earlier, in some situations (such as allocation in the host code) additional annotations will be needed to use these address spaces.  Thus we'll want to allow users to create multi_ptrs in these address spaces so that they can provide more information to the compiler in the same way they can today to disambiguate generic pointers whose address space can't be deduced by the compiler.